src/domain/flush/flush.repository.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Inject,
+  Injectable,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import { CacheRouter } from '../../datasources/cache/cache.router';
+import {
+  CacheService,
+  ICacheService,
+} from '../../datasources/cache/cache.service.interface';
+import { IConfigApi } from '../interfaces/config-api.interface';
+import { InvalidationPatternDto } from './entities/invalidation-pattern.dto.entity';
+import { InvalidationTarget } from './entities/invalidation-target.entity';
+import { IFlushRepository } from './flush.repository.interface';
+
+@Injectable()
+export class FlushRepository implements IFlushRepository {
+  constructor(
+    @Inject(CacheService) private readonly cacheService: ICacheService,
+    @Inject(IConfigApi) private readonly configApi: IConfigApi,
+  ) {}
+
+  async execute(pattern: InvalidationPatternDto): Promise<void> {
+    switch (pattern.invalidate) {
+      case InvalidationTarget.Chains:
+        return this.invalidateChains();
+      case InvalidationTarget.Contracts:
+        return this.invalidateContracts();
+      case InvalidationTarget.Tokens:
+        return this.invalidateTokens(pattern);
+    }
+  }
+
+  private async invalidateChains(): Promise<void> {
+    const chains = await this.configApi.getChains();
+    await this.cacheService.delete(CacheRouter.getChainsCacheDir().key);
+    await Promise.all(
+      chains.results.map(async (chain) =>
+        this.cacheService.delete(
+          CacheRouter.getChainCacheDir(chain.chainId).key,
+        ),
+      ),
+    );
+  }
+
+  private async invalidateContracts(): Promise<void> {
+    const chains = await this.configApi.getChains();
+    await Promise.all(
+      chains.results.map(async (chain) =>
+        this.cacheService.deleteByPattern(
+          CacheRouter.getContractsCachePattern(chain.chainId),
+        ),
+      ),
+    );
+  }
```

Comment:
Here we're invalidating the contracts of each chain one by one. Probably we could want to invalidate contracts for a specific chain selectively in the future, so I included the `chainId` into the cache pattern. What do you think @fmrsabino? If you don't see any value here, we can just invalidate all the contracts at one by using the pattern `*_contract`.